crates/redeem/src/lib.rs

```diff
@@ -77,35 +77,35 @@ pub mod pallet {
             redeem_id: H256,
             redeemer: T::AccountId,
             vault_id: DefaultVaultId<T>,
-            amount: Wrapped<T>,
-            fee: Wrapped<T>,
-            premium: Collateral<T>,
+            amount: BalanceOf<T>,
+            fee: BalanceOf<T>,
+            premium: BalanceOf<T>,
             btc_address: BtcAddress,
-            transfer_fee: Wrapped<T>,
+            transfer_fee: BalanceOf<T>,
         },
         LiquidationRedeem {
             redeemer: T::AccountId,
-            amount: Wrapped<T>,
+            amount: BalanceOf<T>,
```

Comment:
My only issue with this is that the type alias serves a documenting role in cases like this - without this type alias it is less clear whether this amount is the amount of tokens burned, or the collateral received. Probably ok though..